app/code/Magento/ConfigurableProduct/Pricing/Price/ConfigurableRegularPrice.php

```diff
@@ -196,26 +196,38 @@ public function _resetState(): void
     }
 
     /**
-     * Check whether Configurable Product have more than one children products
+     * @deprecated
+     * @see isPriceEqualAcrossChildProducts
+     *
+     * Check whether Configurable options do not have price difference
      *
      * @param SaleableInterface $product
      * @return bool
      */
     public function isChildProductsOfEqualPrices(SaleableInterface $product): bool
+    {
+        return $this->isPriceEqualAcrossChildProducts();
+    }
+
+    /**
+     * Check whether Configurable options do not have price difference
+     *
+     * @return bool
+     */
+    public function isPriceEqualAcrossChildProducts(): bool
     {
         $minPrice = $this->getMinRegularAmount()->getValue();
-        $final_price = $product->getFinalPrice();
-        $productId = $product->getId();
-        if ($final_price < $minPrice) {
+
+        if ($this->product instanceof Product && $this->product->getFinalPrice() < $minPrice) {
             return false;
         }
-        $attributes = $product->getTypeInstance()->getConfigurableAttributes($product);
-        $items = $attributes->getItems();
-        $options = reset($items);
-        $maxPrice = $this->configurableMaxPriceCalculator->getMaxPriceForConfigurableProduct($productId);
-        if ($maxPrice == 0) {
+
+        $maxPrice = $this->configurableMaxPriceCalculator
+            ->getMaxPriceForConfigurableProduct((int)$this->product->getId());
```

Comment:
`            ->getMaxPriceForConfigurableProduct((int) $this->product->getId());
`